gateway/webhook/webhook.go

```diff
@@ -179,25 +177,31 @@ func (webhook *HandleT) RequestHandler(w http.ResponseWriter, r *http.Request) {
 		r.Header.Set("Content-Type", "application/json")
 	}
 
-	done := make(chan webhookErrorRespT)
-	req := webhookT{request: r, writer: &w, done: done, sourceType: sourceDefName, writeKey: writeKey}
+	done := make(chan transformerResponse)
+	req := webhookT{request: r, writer: w, done: done, sourceType: sourceDefName, writeKey: writeKey}
 	webhook.requestQ[sourceDefName] <- &req
 
 	// Wait for batcher process to be done
 	resp := <-done
 	webhook.gwHandle.IncrementAckCount(1)
 	atomic.AddUint64(&webhook.ackCount, 1)
-	webhook.gwHandle.TrackRequestMetrics(resp.err)
-	if resp.err != "" {
+	webhook.gwHandle.TrackRequestMetrics(resp.Err)
+
+	if resp.Err != "" {
 		code := http.StatusBadRequest
-		if resp.statusCode != 0 {
-			code = resp.statusCode
+		if resp.StatusCode != 0 {
+			code = resp.StatusCode
 		}
-		pkgLogger.Infof("IP: %s -- %s -- Response: %d, %s", misc.GetIPFromReq(r), r.URL.Path, code, resp.err)
-		http.Error(w, resp.err, code)
+		pkgLogger.Infof("IP: %s -- %s -- Response: %d, %s", misc.GetIPFromReq(r), r.URL.Path, code, resp.Err)
+		http.Error(w, resp.Err, code)
```

Comment:
[minor] an early return here can make the code more readable.

```suggestion
		http.Error(w, resp.Err, code)
		return
```
(no need for else case)

Reply:
Done